### PR TITLE
Refactor of is Moderator and is Supervisor middleware

### PR DESCRIPTION
**Pull request by jayserdny**
Changes made:

I've merged both middlewares into a single middleware that is able to check any role provided as a parameter to avoid duplicated code.

### CODE REVIEW

**Comment by jayserdny** [comment]
NOTE: Ignore all the commits. They are marked as repeat because this master branch is only squashing the commits.

File: src/api/middlewares/check_role.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const User = require('../models/user.model');
+const httpStatus = require('http-status');
+
+/**
+ * Check the role of the current user
+ * @author Jayser Mendez.
+ * @public
+ */
+// eslint-disable-next-line
+const checkRole = (role) => {
```
Comment (jayserdny) [inline src/api/middlewares/check_role.js:11]: This is the only new thing added basically

Comment (the-codingninja) [reply src/api/middlewares/check_role.js:11]: I can see but wouldn't it be better if you can rebase it on top of master :slightly_smiling_face:

Comment (jayserdny) [reply src/api/middlewares/check_role.js:11]: I did it :/ but all the commits are there again lol

Comment (the-codingninja) [reply src/api/middlewares/check_role.js:11]: okay As this one would be squashed and merge reset your fork to make future better :smile:

Comment (jayserdny) [reply src/api/middlewares/check_role.js:11]: I will do that lol. Idk why this happened hahhaah

**Review by the-codingninja — APPROVED** [review]

**Comment by the-codingninja** [comment]
@jayserdny you can reset your fork now from upstream

**Comment by jayserdny** [comment]
@ms10398 Did it 😎

**Pull request merged**